components/prism-mongodb-query.js

```diff
@@ -0,0 +1,133 @@
+(function (Prism) {
+
+	var keywords = [
+		'$eq',
+		'$gt',
+		'$gte',
+		'$in',
+		'$lt',
+		'$lte',
+		'$ne',
+		'$nin',
+		'$and',
+		'$not',
+		'$nor',
+		'$or',
+		'$exists',
+		'$type',
+		'$expr',
+		'$jsonSchema',
+		'$mod',
+		'$regex',
+		'$text',
+		'$where',
+		'$geoIntersects',
+		'$geoWithin',
+		'$near',
+		'$nearSphere',
+		'$box',
+		'$center',
+		'$centerSphere',
+		'$geometry',
+		'$maxDistance',
+		'$minDistance',
+		'$polygon',
+		'$uniqueDocs',
+		'$all',
+		'$elemMatch',
+		'$size',
+		'$bitsAllClear',
+		'$bitsAllSet',
+		'$bitsAnyClear',
+		'$bitsAnySet',
+		'$comment',
+		'$meta',
+		'$slice',
+		'$currentDate',
+		'$inc',
+		'$min',
+		'$max',
+		'$mul',
+		'$rename',
+		'$set',
+		'$setOnInsert',
+		'$unset',
+		'$addToSet',
+		'$pop',
+		'$pull',
+		'$push',
+		'$pullAll',
+		'$each',
+		'$position',
+		'$slice',
+		'$sort',
+		'$bit',
+	];
+
+	var functions = [
+		'ObjectId',
+		'Code',
+		'BinData',
+		'DBRef',
+		'Timestamp',
+		'NumberLong',
+		'NumberDecimal',
+		'MaxKey',
+		'MinKey',
+		'RegExp',
+		'ISODate',
+		'UUID',
+	];
+
+	keywords = keywords.map(function(keyword) { 
+		return keyword.replace('$', '\\$');
+	});
+
+	var keywordsRegex = '(?:' + keywords.join('(?:\\b|:)|') + ')\\b';
```

Comment:
This will generate a string value like this:

```
(?:\$foo(?:\b|:)|...|\$bar(?:\b|:)|\$baz)\b
```

Assuming that it's a bug that the last keyword doesn't get the `(?:\b|:)` suffix, we can factor out the common pre- and suffixes like this:

```
\$(?:foo|...|bar|baz)(?:\b|:)\b
```

Let's talk about the `(?:\b|:)\b` suffix. It's equivalent to `(?:\b|:\b)` where the problem is easier to see. The `:\b` alternative can never be matched. If we have a string `"$foo:"`, then the `\b` alternative will accept after `"$foo"`.
So we can simplify the whole pattern even further:

```
\$(?:foo|...|bar|baz)\b
```

But we know that the `\b` assertion will always accept because of the way the this pattern is used. It's inserted to create the `keyword` regex, so we know that what follows looks like this: `["']?$`. Since we know that `\b` always accepts, we can just remove it.

```
\$(?:foo|...|bar|baz)
```

^ This is the string we want to generate.
So you can remove the `$` prefix in all strings of the `keywords` array, you can remove the mapping adding a `\` character, and this line becomes:

```suggestion
	var keywordsRegex = '\\$(?:' + keywords.join('|') + ')';
```